ion-schema/src/system.rs

```diff
@@ -948,10 +948,26 @@ impl Resolver {
             );
         }
 
-        Ok(Rc::new(Schema::new(
-            isl.id(),
-            Rc::new(type_store.to_owned()),
-        )))
+        let schema = Arc::new(Schema::new(isl.id(), Arc::new(type_store.to_owned())));
```

Comment:
This function is where `TypeStore` is modified with resolved types. Which is why I am taking in `&mut TypeStore` here. If we use `Arc<TypeStore>`then we won't be able to modify it.

Reply:
Probably I can create a new `TypeStore` in this function itself instead of getting it from `SchemaSystem`. Does that sound good to you?

Reply:
> If we use Arc<TypeStore>then we won't be able to modify it.

Good point. Can we change `to_owned()` to `clone()` here so that it's more clear that we do actually want a copy of the type store?

Reply:
Done. changed with https://github.com/amazon-ion/ion-schema-rust/pull/153/commits/caa4ac21afa3876c1920d23640cf9464853b54d1